libs/shared/contentful/src/lib/queries/purchase-with-details-offering-content/factories.ts

```diff
@@ -0,0 +1,87 @@
+import { faker } from '@faker-js/faker';
+import {
+  OfferingCommonContentResult,
+  PurchaseDetailsResult,
+  PurchaseDetailsTransformed,
+  PurchaseOfferingResult,
+  PurchaseWithDetailsOfferingContentByPlanIdsResult,
+  PurchaseWithDetailsOfferingContentResult,
+  PurchaseWithDetailsOfferingContentTransformed,
+} from './types';
+
+export const PurchaseDetailsResultFactory = (
+  override?: Partial<PurchaseDetailsResult>
+): PurchaseDetailsResult => ({
+  details: `${faker.string.alpha(10)}\n${faker.string.alpha(10)}`,
+  productName: faker.string.alpha(10),
+  subtitle: faker.string.alpha(10),
+  webIcon: faker.internet.url(),
+  ...override,
+});
+
+export const PurchaseDetailsTransformedFactory = (
+  override?: Partial<PurchaseDetailsTransformed>
+): PurchaseDetailsTransformed => ({
+  ...PurchaseDetailsResultFactory(),
+  details: Array.from({ length: faker.number.int({ min: 1, max: 4 }) }, () =>
+    faker.string.alpha(10)
+  ),
+  ...override,
+});
+
+export const OfferingCommonContentResultFactory = (
+  override?: Partial<OfferingCommonContentResult>
+): OfferingCommonContentResult => ({
+  privacyNoticeUrl: faker.internet.url(),
+  privacyNoticeDownloadUrl: faker.internet.url(),
+  termsOfServiceUrl: faker.internet.url(),
+  termsOfServiceDownloadUrl: faker.internet.url(),
+  cancellationUrl: faker.internet.url(),
+  emailIcon: faker.internet.url(),
+  successActionButtonUrl: faker.internet.url(),
+  successActionButtonLabel: faker.string.alpha(10),
+  newsletterLabelTextCode: faker.string.alpha(10),
```

Comment:
Per Ecosystem Docs, the values for newsletterLabelTextCode are: `snp`, `hubs`, `mdnplus`. Should this be `faker.helpers.arrayElements(['snp', 'hubs', 'mdnplus']),`?
Ref: [https://mozilla.github.io/ecosystem-platform/tutorials/subscription-platform#stripe-product-metadata](https://mozilla.github.io/ecosystem-platform/tutorials/subscription-platform#stripe-product-metadata)